examples/fabric-bridge-app/fabric-bridge-common/src/ZCLCallbacks.cpp

```diff
@@ -0,0 +1,92 @@
+#define ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_FEATURE_MAP (0u)
+
+// External attribute read callback function
+Protocols::InteractionModel::Status emberAfExternalAttributeReadCallback(EndpointId endpoint, ClusterId clusterId,
+                                                                         const EmberAfAttributeMetadata * attributeMetadata,
+                                                                         uint8_t * buffer, uint16_t maxReadLength)
+{
+    uint16_t endpointIndex  = emberAfGetDynamicIndexFromEndpoint(endpoint);
+    AttributeId attributeId = attributeMetadata->attributeId;
+
+    Device * dev = DeviceMgr().GetDevice(endpointIndex);
+    if (dev != nullptr && clusterId == app::Clusters::BridgedDeviceBasicInformation::Id)
+    {
+        using namespace app::Clusters::BridgedDeviceBasicInformation::Attributes;
+        ChipLogProgress(NotSpecified, "HandleReadBridgedDeviceBasicAttribute: attrId=%d, maxReadLength=%d", attributeId,
+                        maxReadLength);
+
+        if ((attributeId == Reachable::Id) && (maxReadLength == 1))
+        {
+            *buffer = dev->IsReachable() ? 1 : 0;
+        }
+        else if ((attributeId == NodeLabel::Id) && (maxReadLength == 32))
+        {
+            MutableByteSpan zclNameSpan(buffer, maxReadLength);
+            MakeZclCharString(zclNameSpan, dev->GetName());
+        }
+        else if ((attributeId == ClusterRevision::Id) && (maxReadLength == 2))
+        {
+            uint16_t rev = ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_CLUSTER_REVISION;
+            memcpy(buffer, &rev, sizeof(rev));
+        }
+        else if ((attributeId == FeatureMap::Id) && (maxReadLength == 4))
+        {
+            uint32_t featureMap = ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_FEATURE_MAP;
+            memcpy(buffer, &featureMap, sizeof(featureMap));
+        }
+        else
+        {
+            return Protocols::InteractionModel::Status::Failure;
+        }
+        return Protocols::InteractionModel::Status::Success;
+    }
+
+    return Protocols::InteractionModel::Status::Failure;
+}
+
+// External attribute write callback function
+Protocols::InteractionModel::Status emberAfExternalAttributeWriteCallback(EndpointId endpoint, ClusterId clusterId,
+                                                                          const EmberAfAttributeMetadata * attributeMetadata,
+                                                                          uint8_t * buffer)
+{
+    uint16_t endpointIndex                  = emberAfGetDynamicIndexFromEndpoint(endpoint);
+    Protocols::InteractionModel::Status ret = Protocols::InteractionModel::Status::Failure;
+
+    Device * dev = DeviceMgr().GetDevice(endpointIndex);
+    if (dev != nullptr && dev->IsReachable())
+    {
+        ChipLogProgress(NotSpecified, "emberAfExternalAttributeWriteCallback: ep=%d, clusterId=%d", endpoint, clusterId);
+        ret = Protocols::InteractionModel::Status::Success;
```

Comment:
How is this a success? You are not writing anything here? Is this even a writeable attribure, should we give an access error here?

I think at this point you are always expected to be done so you can return the value here instead of setting ret otherwise later on you might end up with different unexpected execution after this

Reply:
Some endpoints of the bridge represent the bridged device. Therefore, all reads or writes to the attributes must be handled within the emberAfExternalAttributeWriteCallback and emberAfExternalAttributeReadCallback functions declared herein. This fits the typical model of a bridge, as a bridge typically maintains its own state database representing the devices connected to it.

If the bridge receives a write operation targeting one of its bridged devices and the device is reachable, we should intercept this operation here. Otherwise, the write operation will go to the bridge itself at that endpoint.

Currently, we just print out a message and do nothing. After all FS clusters have landed, we might need to forward the write operation to the bridged device.